src/js/messaging/reducers/modals.js

```diff
@@ -26,6 +29,8 @@ export default function modals(state = initialState, action) {
       return set('attachments.visible', !state.attachments.visible, state);
     case TOGGLE_CREATE_FOLDER:
       return set('createFolder.visible', !state.createFolder.visible, state);
+    case SET_FOLDER_NAME:
+      return set('createFolder.folderName', action.folderName, state);
```

Comment:
Path also needs to be updated to `createFolder.newFolderName`. I guess also update `SET_FOLDER_NAME` -> `SET_NEW_FOLDER_NAME`.